core-bundle/src/Resources/contao/library/Contao/Search.php

```diff
@@ -782,33 +783,31 @@ public static function query(string $strKeywords, bool $blnOrSearch=false, array
 	 *
 	 * @param string $strUrl The URL to be removed
 	 */
-	public static function removeEntry($strUrl)
+	public static function removeEntry($strUrl, Connection $connection = null)
 	{
-		$objDatabase = Database::getInstance();
+		$connection = $connection ?? System::getContainer()->get('database_connection');
 
-		$objResult = $objDatabase->prepare("SELECT id FROM tl_search WHERE url=?")
-								 ->execute($strUrl);
+		$result = $connection
+			->prepare('SELECT id FROM tl_search WHERE url=:url')
+			->executeQuery(array('url' => $strUrl));
 
-		while ($objResult->next())
+		foreach ($result->fetchAllAssociativeIndexed() as $id)
 		{
 			// Decrement document frequency counts
-			$objDatabase
+			$connection
 				->prepare("
 					UPDATE tl_search_term
-					INNER JOIN tl_search_index ON tl_search_term.id = tl_search_index.termId AND tl_search_index.pid = ?
+					INNER JOIN tl_search_index ON tl_search_term.id = tl_search_index.termId AND tl_search_index.pid = :id
```

Comment:
```suggestion
					INNER JOIN tl_search_index ON tl_search_term.id = tl_search_index.termId AND tl_search_index.pid = ?
```

Using `?` instead of the named parameters fixed in in my setup.